Align 'gitlab_runner' with GitLab's default access level.

cfiehe opened the pull request:
##### SUMMARY
In order to align the module with GitLab's runner API, the option ``access_level`` will lose its default value in community.general 8.0.0. From that version on, you have set this option to ``ref_protected`` explicitly, if you want to have a protected runner, otherwise GitLab's default access level gets applied, which is ``not_protected``.

Because changing the default value requires a longer deprecation period, a deprecation message was added, that informs the user about the future change regarding module's behavior.

Signed-off-by: Christoph Fiehe <c.fiehe@eurodata.de>

##### ISSUE TYPE
<!--- Pick one or more below and delete the rest.
      'Test Pull Request' is for PRs that add/extend tests without code changes. -->
- Feature Pull Request

##### COMPONENT NAME
gitlab_runner

cfiehe commented:
@nejch @felixfontein I hope, that is what you have in mind with respect to changing the default value of the option ``access_level``.

felixfontein commented on plugins/modules/gitlab_runner.py:
```diff
@@ -86,11 +86,10 @@
         is only applied on updates.
-      - If set to C(ref_protected), runner can pick up jobs only from protected branches.
       - If set to C(not_protected), runner can pick up jobs from both protected and unprotected branches.
+      - If set to C(ref_protected), runner can pick up jobs only from protected branches.
```
This should say something like:
```suggestion
      - If set to C(ref_protected), runner can pick up jobs only from protected branches.
      - The current default is C(ref_protected). This will change to no default in community.general 8.0.0.
        From then on, there will be no default. When this option is not specified, GitLab will use C(not_protected)
        on creation, and the value set for the runner will not be changed on updates.
```

cfiehe replied:
Thanks a lot. I have extended the documentation section accordingly.

felixfontein reviewed (commented):
Looks good to me. One thing I'm wondering: to implement the removal of the default, are more code changes needed than simply removing the deprecation warning and the assignment, and updating the docs? (Like making sure that `None` is handled correctly by not passing it to the API?)

nejch commented:
> Looks good to me. One thing I'm wondering: to implement the removal of the default, are more code changes needed than simply removing the deprecation warning and the assignment, and updating the docs? (Like making sure that `None` is handled correctly by not passing it to the API?)

I'd need to test this specifically, as GitLab can vary depending on the endpoint. But I'd say that yes, generally we should only collect non-null attributes where gitlab expects a value, to be safer. I'll try to get to this PR tomorrow unless @cfiehe  you have time to test POST/PUT against this endpoint with a null attribute passed to it

cfiehe commented:
I have made a quick test and passed `access_level` to the API with a `None` value. It does not seem not cause any problems. But I would prefer to pass `access_level` to the server only in case when it is not `None`. Yes, that requires a code change, but it is only a small one. I think its worth 🙂.

cfiehe commented:
I have modified the pull request. I think, we can do it the other way round. The `access_level` is added to the arguments dictionary if it is not `None` `(the future behavior) and removed again on runner creation if `access_level_on_creation` is `False`.

felixfontein reviewed (commented):
Looks good from my POV!

nejch reviewed (approved):
LGTM! maybe i'd do `s/loose/lose/` but doesn't matter it's just text ;)

cfiehe commented:
Upps! The typo has been fixed 😇. Thanks a lot.

cfiehe commented:
recheck

cfiehe commented:
recheck

felixfontein commented:
@cfiehe thanks for your contribution!
@nejch thanks for reviewing this!
